packages/riipen-ui/src/components/Popover.jsx

```diff
@@ -274,140 +120,145 @@ class Popover extends React.Component {
           );
           closestScrollableParent.addEventListener(
             "scroll",
-            closeOnScrolled ? this.handleClose : this.positionChangeEventHandler
+            closeOnScrolled ? handleClose : positionChangeEventHandler
           );
         }
       }
     }
-  }
-
-  isDecendant(parent, child) {
-    const node = child.parentNode;
-    if (node == null) {
-      return false;
-    } else if (node === parent) {
-      return true;
-    }
-    return this.isDecendant(parent, node);
-  }
-
-  handleCloseEvent = event => {
-    const { isOpen, anchorEl } = this.props;
-    const contentRef = this.contentRef.current;
-    if (!contentRef || !isOpen) return;
-    if (event.type === "mousedown") {
-      // Check if mouse click happened inside popover
-      // Or on the anchorEl
-      const target = event.target;
-      if (
-        contentRef === target ||
-        anchorEl === target ||
-        this.isDecendant(contentRef, target) ||
-        this.isDecendant(anchorEl, target)
-      ) {
-        return;
-      }
-      this.handleClose();
-    } else if (event.type === "keydown" && event.key === "Escape") {
-      this.handleClose();
-    } else if (event.type === "keydown" && this.props.onKeyDown) {
-      // stop this event from triggering other keydown listeners
-      event.stopImmediatePropagation();
-
-      this.props.onKeyDown(event);
-    }
   };
 
-  handleClose = () => {
-    const { onClose } = this.props;
-    if (onClose) onClose();
-  };
+  useEffect(() => {
+    window.addEventListener("resize", positionChangeEventHandler);
+    window.addEventListener("scroll", positionChangeEventHandler);
+    window.addEventListener("mousedown", handleCloseEvent);
+    window.addEventListener("keydown", handleCloseEvent);
 
-  // Find the outer most scrollable content and remove the scrollbar
-  lockParentScroll() {
-    const anchorEl = this.getAnchorEl();
-    const parent = anchorEl.parentElement;
-    const container = getDocument(anchorEl);
+    if (anchorEl) {
+      manageScrollableParentEventListeners(false);
+    }
 
-    this.state.scrollContainer =
-      parent.nodeName === "HTML" &&
-      window.getComputedStyle(parent)["overflow-y"] === "scroll"
-        ? parent
-        : container.body;
+    setPositioningStyle();
 
-    this.state.scrollContainerStyle = this.state.scrollContainer.style.overflow;
-    this.state.scrollContainer.style.overflow = "hidden";
-  }
+    return () => {
+      window.removeEventListener("resize", positionChangeEventHandler);
+      window.removeEventListener("scroll", positionChangeEventHandler);
+      window.removeEventListener("mousedown", handleCloseEvent);
+      window.removeEventListener("keydown", handleCloseEvent);
 
-  resetParentScroll() {
-    this.state.scrollContainer.style.overflow = this.state.scrollContainerStyle;
-  }
+      manageScrollableParentEventListeners(true);
+    };
+  }, []);
+
+  useEffect(() => {
```

Comment:
I think it would be nice if we leave comments for each useEffect. For example we can copy this comment from the previous Popover component.
```
// Find the outer most scrollable content and remove the scrollbar
```